Source/ARTPushActivationStateMachine.m

```diff
@@ -85,6 +85,15 @@ - (ARTPushActivationState *)current_nosync {
     return _current;
 }
 
+- (ARTLocalDevice *)localDevice {
+    ARTLocalDevice *device = _rest.device_nosync;
+    if (device.id == nil) {
+        [_rest resetDeviceSingleton];
+        device = _rest.device_nosync;
+    }
+    return device;
+}
```

Comment:
Should this logic not live with the rest of the device information (inside `ARTRest`)? It feels out of place here, when the state is inside `_rest`. We use `_rest.device_nosync` elsewhere in the code, and now there are 2 ways (and 2 "properties") for people to use. IMHO we should only have 1

Reply:
Good idea, will now move logic from new `localDevice` method to old getter `device_nosync`.

Reply:
Actually not a good idea, there was a reason why they were separated - you can't recreate `ARTLocalDevice.shared_nosync` each time you call `device_nosync` even if `ARTLocalDevice.id` is `nil`.

Reply:
Both methods are private, so if you refer to people as ably-cocoa contributors, then one should be aware what method to use - the one that recreates or the one that returns as-is.

Reply:
Log messages are not related to ably-cocoa IMHO.

Reply:
Can/shoud this be documented? It's not obvious _to me_, and I'm afraid of the harm and friction this will do for future developers. Having said that, I am not an objective-C developer.

Reply:
I've now removed all calls where device can be created except two places where it's actually should be created. Now it's explicitly, except `device` property, which is heavily used in tests, so I left it for now until API settles down. Then I'll fix tests so they don't use this property as well. @ben-xD

Reply:
b18bc1ae